### PR TITLE
migration to add more model meta

### PR DESCRIPTION
**Pull request by xiangli313**
YoutracK:
https://vimc.myjetbrains.com/youtrack/issue/VIMC-2361
This migration should go together with data import PR https://github.com/vimc/montagu-data/pull/107.

### CODE REVIEW

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  is_pine BOOLEAN NOT NULL, 
+  PRIMARY KEY (id),
+  FOREIGN KEY (model_version) REFERENCES model_version(id),
+  FOREIGN KEY (country) REFERENCES country(id)
+);
+COMMENT ON TABLE model_version_country IS
+  'Specify which countries are evaluated in each model, and which countries are considered in small-scale model run.';
+
+
+
```
Comment (richfitz) [inline migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: I think we can safely delete the columns `model_version.note` and `model_version.fingerprint` here - they contain no sensible information at the moment

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+--- 3. model_version_country
+
+COMMENT ON COLUMN model.is_current IS
+  'TRUE if the model is currently running for VIMC.';
+
+CREATE TABLE model_version_meta (
+  id SERIAL,
+  model_version INTEGER NOT NULL,
+  is_dynamic BOOLEAN NOT NULL, 
+  code TEXT,
```
Comment (richfitz) [inline migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: should `code` be `language`?

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  dalys BOOLEAN NOT NULL,
```
Comment (richfitz) [inline migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: `computes_dalys`?

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  gender INTEGER NOT NULL,
+  burden_min_age INTEGER NOT NULL,
+  burden_max_age INTEGER NOT NULL,
+  cohort_first INTEGER, --- nullable, NA means 2000
```
Comment (richfitz) [inline migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: Why nullable meaning 2000? why not set to 2000 in the import? If that's for some reason not possible or desirable please explain and then document the NA meaning in the comment for the field (and the same for all other columns with "meaningful" NA values)

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  cohort_last INTEGER, --- nullable, NA means min(2100, age_max+2030)
+  year_min INTEGER, --- nullable, NA means 2000
```
Comment (richfitz) [inline migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: we have both `_first`/`_last` and `_min`/`_max` here. Would you like to standardise on one?

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+);
+COMMENT ON TABLE model_version_meta IS
+  'General model metadata, including whether models consider herd effect, how models are coded, whether models provide DALYs, model observation shape - age/cohort/year ranges.';
+
+CREATE TABLE model_version_vaccine (
+  id SERIAL,
+  model_version INTEGER NOT NULL,
+  vaccine TEXT NOT NULL, 
+  activity_type TEXT NOT NULL, 
+  delivery_order INTEGER NOT NULL,
```
Comment (richfitz) [inline migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: in the import script you use `order` - please update one or the other.  Does this not risk falling into some tension with the same concept in the responsibilities (via `secenario_coverage_set`?  We don't want the same information duplicated generally.

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  FOREIGN KEY (activity_type) REFERENCES activity_type(id)
+);
+COMMENT ON TABLE model_version_vaccine IS
+  'Specify which vaccine deliveries (in delivery order) are evaluated in each model.';
+
+CREATE TABLE model_version_country (
+  id SERIAL,
+  model_version INTEGER NOT NULL,
+  country TEXT NOT NULL, 
+  is_pine BOOLEAN NOT NULL,
```
Comment (richfitz) [inline migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: Please drop this column - we can get this (and should) by joining through to the country metadata that you created already!

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql (hunk 1 above)
Comment (xiangli313) [reply migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: Done.

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql (hunk 2 above)
Comment (xiangli313) [reply migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: Done.

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql (hunk 3 above)
Comment (xiangli313) [reply migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: Done.

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql (hunk 5 above)
Comment (xiangli313) [reply migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: Done. All with `_min/_max`.

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql (hunk 6 above)
Comment (xiangli313) [reply migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: OK. I have removed order/delivery_order. It dose duplicate order in scenario_coverage_set.

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql (hunk 7 above)
Comment (xiangli313) [reply migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: You mean drop column - country. I agree. 
I am changing the table to 
```
id,
model_version,
pine_country
```

File: migrations/sql/V2018.11.20.1106__AddModelMeta.sql (hunk 4 above)
Comment (xiangli313) [reply migrations/sql/V2018.11.20.1106__AddModelMeta.sql]: I am making them explicit. Done.

**Comment by hillalex** [comment]
This PR has been open since 2018 - should it be closed? @richfitz

**Comment by richfitz** [comment]
Probably, yeah. There is I presume a data import that goes along with it. @xiangli313 - do you remember anything about this?

**Comment by xiangli313** [comment]
@richfitz and @hillalex. I think I remember. In the first place, we wanted to import unique truth of model information into the db, instead of keeping them as csvs. Then I thought it might be useful to link model/model version to their own burden outcomes and year/cohort/age/country constrains. It should be easier to prepare burden templates. I think we still want to do this.

**Comment by richfitz** [comment]
Thanks Xiang - I will bring this up to date, but can you confirm that we still have the correct fields here (nothing to add, nothing to remove).  the import will need translating to use dettl but that should not be too hard

File: migrations/sql/V2020.02.12.1100__AddModelMeta.sql [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+COMMENT ON COLUMN model.is_current IS
+  'TRUE if the model is currently running for VIMC.';
+
+CREATE TABLE model_version_meta (
+  id SERIAL,
+  model_version INTEGER NOT NULL,
+  is_dynamic BOOLEAN NOT NULL, 
+  language TEXT,
+  computes_dalys BOOLEAN NOT NULL,
+  gender INTEGER NOT NULL,
```
Comment (richfitz) [inline migrations/sql/V2020.02.12.1100__AddModelMeta.sql]: In V2019.07.04.1441__AddModelMetadata.sql we added `gender` to the `model` table.  Will we need to remove that at some point?

File: migrations/sql/V2020.02.12.1100__AddModelMeta.sql [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+	DROP COLUMN note,
+	DROP COLUMN fingerprint;
+
+COMMENT ON COLUMN model.is_current IS
+  'TRUE if the model is currently running for VIMC.';
+
+CREATE TABLE model_version_meta (
+  id SERIAL,
+  model_version INTEGER NOT NULL,
+  is_dynamic BOOLEAN NOT NULL,
```
Comment (richfitz) [inline migrations/sql/V2020.02.12.1100__AddModelMeta.sql]: is_dynamic (and code) were added to the table model_version earlier - do we want to do something about that too?

**Pull request closed without merging**